check_diff/src/lib.rs

```diff
@@ -56,3 +162,146 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
     );
     return Ok(());
 }
+
+pub fn get_ld_lib_path() -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new("rustc").args(["--print", "sysroot"]).output() else {
+        return Err(CheckDiffError::FailedCommand(
+            "Error getting sysroot".to_string(),
+        ));
+    };
+
+    let Ok(sysroot) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting sysroot to string".to_string(),
+        ));
+    };
+
+    let ld_lib_path = format!("{}/lib", sysroot.trim_end());
+    return Ok(ld_lib_path);
+}
+
+pub fn get_cargo_version() -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new("cargo").args(["--version"]).output() else {
+        return Err(CheckDiffError::FailedCargoVersion(
+            "Failed to obtain cargo version".to_string(),
+        ));
+    };
+
+    let Ok(cargo_version) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting cargo version to string".to_string(),
+        ));
+    };
+
+    return Ok(cargo_version);
+}
+
+pub fn get_binary_version(
+    binary: &PathBuf,
```

Comment:
Prefer to use `&Path` instead of `&PathBuf`. `&Path` or even `AsRef<Path>` is a more flexible argument type for callers. I'd double check and update all function / method signatures that use `&PathBuf`.